synapse/rest/media/v1/preview_url_resource.py

```diff
@@ -268,6 +271,17 @@ def _do_preview(self, url, user, ts):
             logger.warn("Failed to find any OG data in %s", url)
             og = {}
 
+        # filter out any stupidly long values
+        keys_to_remove = []
+        for k, v in og.items():
+            if len(k) > OG_TAG_NAME_MAXLEN or len(v) > OG_TAG_VALUE_MAXLEN:
+                logger.warning(
+                    "Pruning overlong tag %s from OG data", k[:OG_TAG_NAME_MAXLEN]
+                )
+                keys_to_remove.append(k)
```

Comment:
Why remove the whole thing if the value (I'm not talking about the key here) is too long rather than truncating the value, e.g. replacing "this is a super long title" with "this is a sup..."? I think the latter would allow clients to offer a better UX.

Reply:
per #synapse-dev: agreed to merge and worry about this later.